pkg/server/router.go

```diff
@@ -48,6 +48,15 @@ func router(ctx context.Context, config *Config, cfg *cmds.Server) http.Handler
 	authed.Path(prefix + "/server-ca.crt").Handler(fileHandler(serverConfig.Runtime.ServerCA))
 	authed.Path(prefix + "/config").Handler(configHandler(serverConfig, cfg))
 	authed.Path(prefix + "/readyz").Handler(readyzHandler(serverConfig))
+	authed.Path(prefix + "/encrypt-status").Handler(encryptionStatusHandler(serverConfig))
+	authed.Path(prefix + "/encrypt-enable").Handler(encryptionToggleHandler(serverConfig, true))
+	authed.Path(prefix + "/encrypt-disable").Handler(encryptionToggleHandler(serverConfig, false))
+	authed.Path(prefix + "/encrypt-prepare").Handler(encryptionPrepareHandler(serverConfig, false))
+	authed.Path(prefix + "/encrypt-prepare-force").Handler(encryptionPrepareHandler(serverConfig, true))
+	authed.Path(prefix + "/encrypt-rotate").Handler(encryptionRotateHandler(serverConfig, false))
+	authed.Path(prefix + "/encrypt-rotate-force").Handler(encryptionRotateHandler(serverConfig, true))
+	authed.Path(prefix + "/encrypt-reencrypt").Handler(encryptionReencryptHandler(serverConfig, false))
+	authed.Path(prefix + "/encrypt-reencrypt-force").Handler(encryptionReencryptHandler(serverConfig, true))
```

Comment:
This isn't very restful. There should be one maybe two endpoints here. Honestly I think just `/encrypt/status`. GETing it returns the current status, and PUTing a chunk of JSON (serialized struct) to it describing what you want to change (enable/disable/prepare/rotate/reencrypt/etc, plus a Force boolean if necessar) to move it through the steps.

Reply:
Just to expand on this, the Kubernetes way to do this would probably be to have a typed struct with a Spec describing the desired state (enabled/disabled, process through the rotation steps, etc) and a Status describing the state of any recent changes. You might talk to @Oats87 about the rancher-system-agent Plan for an example.

Reply:
Converted to a PUT command with a struct body, now only 3 endpoints: status, toggle, stage